pandas/tests/groupby/test_groupby.py

```diff
@@ -2059,6 +2060,49 @@ def test_groups_repr_truncates(max_seq_items, expected):
         assert result == expected
 
 
+@td.skip_if_no("lxml")
+@pytest.mark.parametrize(
+    "n_groups,n_rows,check_n_groups,check_n_rows",
+    [
+        (10, 60, 5, 3),  # All groups and all rows in the groups are shown
+        (25, 100, 5, 2),  # Not all groups are shown
+        (4, 400, 2, 7),  # Not all rows are shown in the groups
+        (20, 400, 5, 3),  # Not all groups and not all rows in the groups are shown
+    ],
+)
+def test_groupby_repr_not_all_groups(n_groups, n_rows, check_n_groups, check_n_rows):
+    # GH 34926
+    df = DataFrame(
+        {
+            "A": range(n_rows),
+            "B": range(0, n_rows * 2, 2),
+            "C": list(range(n_groups)) * (n_rows // n_groups),
+        }
+    )
+
+    df_groupby = df.groupby("C")
```

Comment:
Personal opinion (feel free to ignore) but I find these names confusing. I'd use `gb` for `df_groupby`, `gb_html` for `html_groupby`, `gb_html_as_frame` for `df_from_html`, `gb_iter` for `df_group_iter`